cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,73 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
-describe('Settings page', () => {
-  before(() => {
+import SettingPageObject from '../support/pages/setting.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+import ProfilePageObject from '../support/pages/profile.pageObject';
+import SignInPageObject from '../support/pages/signIn.pageObject';
 
-  });
+const settingPage = new SettingPageObject();
+const homePage = new HomePageObject();
+const profilePage = new ProfilePageObject();
+const signInPage = new SignInPageObject();
+
+describe('Settings page', () => {
+  let user;
+  let newUserName;
+  let newEmail;
+  let newPassword;
 
   beforeEach(() => {
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+      newUserName = user.username + '134';
+      newEmail = 'new' + user.email;
+      newPassword = 'new' + user.password;
+      cy.register(user.email, user.username, user.password);
+      cy.login(user);
+    });
 
+    cy.visit('/settings');
   });
 
   it('should provide an ability to update username', () => {
-
+    settingPage.typeUserName(newUserName);
+    settingPage.clickUpdateButton();
+    homePage.clickOnUsernameLinkInHeader();
+    homePage.assertHeaderContainUsername(newUserName);
   });
 
   it('should provide an ability to update bio', () => {
-
+    settingPage.typeBio(user.username);
+    settingPage.clickUpdateButton();
+    homePage.clickOnUsernameLinkInHeader();
+    profilePage.assertBioContainNewBio(user.username);
   });
 
   it('should provide an ability to update an email', () => {
-
+    settingPage.typeEmail(newEmail);
+    settingPage.clickUpdateButton();
+    settingPage.clickLogoutButton();
+    signInPage.visit('user/login');
+    signInPage.typeEmail(newEmail);
+    signInPage.typePassword(user.password);
```

Comment:
When updating the email, ensure that the password used for login is the correct one associated with the new email. If the password is also updated, use `newPassword` instead of `user.password`.